src/python/controller/controller.py

```
@@ -709,6 +754,23 @@ def __update_model(self):
             self.__persist.extract_failed_file_names.add(fail_key)
             self._sync_persist_to_all_builders()
 
+        # Process validation completions — mark as validated
+        for result in latest_validated_results:
+            self.logger.info("Validation passed for '{}'".format(result.name))
```

Ruff (0.15.5)

[warning] Use f-string instead of `format` call

Convert to f-string

(UP032)